src/BasicVault.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "yield-utils-v2/token/IERC20.sol";
+
+/**
+@title An on-chain name registry
+@author Calnix
+@dev Vault for a specific ERC20 token; token address to be passed to constructor.
+@notice Vault for a pre-defined ERC20 token that was set on deployment. 
+*/
+
+contract BasicVault {
+    
+    ///@notice ERC20 interface specifying token contract functions
+    ///@dev For constant variables, the value has to be fixed at compile-time, while for immutable, it can still be assigned at construction time.
+    IERC20 public immutable wmdToken;    
+
+    ///@notice mapping addresses to their respective token balances
+    mapping(address => uint) public balances;
+
+    /// @notice Emit event when tokens are deposited into Vault
+    event Deposit(address indexed, uint amt);
```

Comment:
You can rename it to `Deposited`. Since, the event has already happened we prefer to keep it in past tense. You can do it for other event as well.